core-processor/src/processing.rs

```diff
@@ -215,15 +214,24 @@ pub fn process_error(
 
     if !dispatch.is_error_reply() && dispatch.kind() != DispatchKind::Signal {
         // This expect panic is unreachable, unless error message is too large or max payload size is too small.
-        let err_payload = err.encode().try_into().expect("Error message is too large");
+        let err_payload = err
+            .to_string()
+            .into_bytes()
+            .try_into()
+            .expect("Error message is too large");
```

Comment:
supposed to be unreachable.
we can like check all of the err variant by some test?

Reply:
It is possible for simple errors but `ActorExecutionErrorReason`